hy/inspect.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2018 the authors.
+# This file is part of Hy, which is free software licensed under the Expat
+# license. See the LICENSE.
+
+from __future__ import absolute_import
+
+import inspect
+
+try:
+    # Check if we have the newer inspect.signature available.
+    # Otherwise fallback to the legacy getargspec.
+    inspect.signature  # noqa
+
+    def get_arity(fn):
+        parameters = inspect.signature(fn).parameters
+        return sum(1 for param in parameters.values()
+                   if param.kind == param.POSITIONAL_OR_KEYWORD)
+
+    def has_kwargs(fn):
+        parameters = inspect.signature(fn).parameters
+        return any(param.kind == param.VAR_KEYWORD
+                   for param in parameters.values())
+
+    def format_args(fn):
+        return str(inspect.signature(fn))
+
+except AttributeError:
+    def get_arity(fn):
+        return len(inspect.getargspec(fn)[0])
+
+    def has_kwargs(fn):
+        argspec = inspect.getargspec(fn)
+        return argspec.keywords is not None
+
+    def format_args(fn):
+        argspec = inspect.getargspec(fn)
+        return inspect.formatargspec(*argspec)
```

Comment:
I think the `try` includes too much code. It would catch an `AttributeError` raised anywhere in those three functions.

Reply:
Or maybe not; maybe it's only in effect when the functions are defined, not called. Still, the code looks confusing in that way, so it seems ideal to just include that first statement in the `try` block.

Reply:
Easy solution: put everything after the first statment in an `else` block.

Reply:
Done

Reply:
I don't see a change. Did you forget to push?

Reply:
Yeah... ssh key wasn't loaded and didn't notice.